paasta_tools/kubernetes_tools.py

```diff
@@ -1865,13 +1868,90 @@ def update_stateful_set(
     )
 
 
+def get_events_for_object(
+    kube_client: KubeClient,
+    obj: Union[V1Pod, V1Deployment, V1StatefulSet, V1ReplicaSet],
+    kind: str,  # for some reason, obj.kind isn't populated when this function is called so we pass it in by hand
+) -> List[V1Event]:
+    host = KubeConfiguration().host
+
+    # The python kubernetes client doesn't support the V1Events API
+    # yet, so we have to make the request by hand (we need the V1Events
+    # API so that we can query by the involvedObject.name/kind)
+    events = kube_client.request(
```

Comment:
Seems like this is the same thing as list_namespaced_event()? https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#list_namespaced_event

Reply:
As best as I can tell, the one under the CoreV1 API does not actually return the "correct" event list.  I did a bunch of experiments with this and I couldn't get any events back, despite events showing up in the event stream.

Reply:
I updated the comment to this effect.